Code_Exercises/Exercise_04_Handling_Errors/solution.cpp

```diff
@@ -27,26 +23,18 @@ TEST_CASE("handling_errors", "handling_errors_source") {
 
     auto defaultQueue = sycl::queue{asyncHandler};
 
-    {
-      auto bufA = sycl::buffer{&a, sycl::range{1}};
-      auto bufB = sycl::buffer{&b, sycl::range{1}};
-      auto bufR = sycl::buffer{&r, sycl::range{1}};
-
-      defaultQueue
-          .submit([&](sycl::handler& cgh) {
-            auto accA = sycl::accessor{bufA, cgh, sycl::read_only};
-            auto accB = sycl::accessor{bufB, cgh, sycl::read_only};
-            auto accR = sycl::accessor{bufR, cgh, sycl::write_only};
+    auto buf = sycl::buffer<int>(sycl::range{1});
 
-            cgh.single_task<scalar_add>([=]() { accR[0] = accA[0] + accB[0]; });
-          })
-          .wait();
-    }
+    defaultQueue.submit([&](sycl::handler& cgh) {
+      // This throws an exception: an accessor has a range which is
+      // outside the bounds of its buffer.
+      auto acc = buf.get_access(cgh, sycl::range{2}, sycl::read_write);
+    }).wait();
 
```

Comment:
```suggestion
printf("This should print before an exception is thrown!\n");
```
I think this would make this exercise very instructive. Would be great if this could be added in the source.cpp as well

Reply:
Actually since the exception is synchronous, surely this won't print at all?

Reply:
Correct, so this should be replaced with something that will cause an async error. I think a wildly out of bounds access is probably the most backend portable

Reply:
```
 defaultQueue.submit([&](sycl::handler &cgh) {
      sycl::accessor acc{buf, cgh};
      cgh.single_task([=]() { acc[INT_MAX] = 6; });
    });

    printf("This should print before we get an exception!\n");
    defaultQueue.wait_and_throw();
```